File: update_nav.js
const fs = require('fs');
const path = require('path');

const dirPath = "c:/Users/dines/OneDrive/Desktop/education-platform-main";

function getAllHtmlFiles(dir, files = []) {
    const list = fs.readdirSync(dir);
    for (const file of list) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            // exclude .vscode, css, img, js directories
            if (['.vscode', 'css', 'img', 'js'].includes(file)) continue;
            getAllHtmlFiles(fullPath, files);
        } else if (file.endsWith('.html')) {
            files.push(fullPath);
        }
    }
    return files;
}

const htmlFiles = getAllHtmlFiles(dirPath);

for (const filepath of htmlFiles) {
    let content = fs.readFileSync(filepath, 'utf8');
    let originalContent = content;
    
    // Determine relative path
    const isPagesDir = filepath.toLowerCase().includes(path.sep + 'pages' + path.sep) || filepath.replace(/\\/g, '/').includes('/pages/');
    const prefix = isPagesDir ? "" : "pages/";
    
    // Dropdown replacement
    const newLinks = `
                        <a href="${prefix}login.html" class="block px-6 py-3 text-sm font-bold text-slate-700 hover:bg-slate-50 hover:text-primary-blue transition">Login/Signup</a>
                        <a href="${prefix}admin-dashboard.html" class="block px-6 py-3 text-sm font-bold text-slate-700 hover:bg-slate-50 hover:text-primary-blue transition">Admin Dashboard</a>
                        <a href="${prefix}user-dashboard.html" class="block px-6 py-3 text-sm font-bold text-slate-700 hover:bg-slate-50 hover:text-primary-blue transition">User Dashboard</a>
`;
    
    // Replace dropdown contents
    const regex = /(<div\s+id="user-dropdown"[^>]*>)([\s\S]*?)(<\/div>)/i;
    content = content.replace(regex, (match, p1, p2, p3) => p1 + newLinks + '                    ' + p3);
    
    // specific changes for login.html
    if (filepath.endsWith('login.html')) {
        content = content.replace(/placeholder="name@email.com"/g, 'placeholder="Email Address"');
        content = content.replace(/placeholder="••••••••"/g, 'placeholder="Password"');
        
        // ensure no values are left from previously hardcoded credentials (if any exist)
        content = content.replace(/value="admin@gmail.com"/g, 'value=""');
        content = content.replace(/value="admin123"/g, 'value=""');
        content = content.replace(/value="user@gmail.com"/g, 'value=""');
        content = content.replace(/value="user123"/g, 'value=""');
    }
    
    if (content !== originalContent) {
        fs.writeFileSync(filepath, content, 'utf8');
        console.log(`Updated: ${filepath}`);
    }
}

console.log("All done!");

File: update_nav.py
import os
import re

dir_path = "c:/Users/dines/OneDrive/Desktop/education-platform-main"

html_files = []
for root, dirs, files in os.walk(dir_path):
    for f in files:
        if f.endswith(".html"):
            html_files.append(os.path.join(root, f).replace("\\", "/"))

def replacer(m, filepath):
    # Determine correct relative path for links
    prefix = "" if "/pages/" in filepath.lower() else "pages/"
    
    new_links = f"""
                        <a href="{prefix}login.html" class="block px-6 py-3 text-sm font-bold text-slate-700 hover:bg-slate-50 hover:text-primary-blue transition">Login/Signup</a>
                        <a href="{prefix}admin-dashboard.html" class="block px-6 py-3 text-sm font-bold text-slate-700 hover:bg-slate-50 hover:text-primary-blue transition">Admin Dashboard</a>
                        <a href="{prefix}user-dashboard.html" class="block px-6 py-3 text-sm font-bold text-slate-700 hover:bg-slate-50 hover:text-primary-blue transition">User Dashboard</a>
"""
    return m.group(1) + new_links + "                    " + m.group(3)

for filepath in html_files:
    with open(filepath, "r", encoding="utf-8") as f:
        content = f.read()
    
    new_content = content
    
    # 1. Update dropdowns
    pattern = re.compile(r'(<div\s+id="user-dropdown"[^>]*>)(.*?)(</div>)', re.DOTALL)
    new_content = pattern.sub(lambda m: replacer(m, filepath), new_content)

    # 2. Update login.html inputs if it's the login.html file
    if filepath.endswith("login.html"):
        new_content = new_content.replace('placeholder="name@email.com"', 'placeholder="Email Address"')
        new_content = new_content.replace('placeholder="••••••••"', 'placeholder="Password"')

        # remove any default selected values if they exist (just in case they were added dynamically or directly)
        new_content = new_content.replace('value="admin@gmail.com"', 'value=""')
        import re
        new_content = re.sub(r'value=".*?"', lambda m: m.group(0) if m.group(0) not in ['value="admin"', 'value="user"'] else m.group(0), new_content) # don't break the <select> option values

    if content != new_content:
        with open(filepath, "w", encoding="utf-8") as f:
            f.write(new_content)
        print("Updated:", filepath)

print("All done!")
